### PR TITLE
Add PreLoadedPlugin mixin for pre-initialized plugins

### PR DESCRIPTION
**Pull request by vishal-kr-barnwal**
This will help us to mark any plugin to be preloaded without any Change to Vyuh Core.

Revamped the handling of preloaded plugins by introducing the PreLoadedPlugin mixin. Applied this mixin to DIPlugin, AnalyticsPlugin, and NetworkPlugin to ensure they are initialized before the platform. Removed the static list of preloadedPlugins from VyuhPlatform.

### CODE REVIEW

File: packages/system/vyuh_core/lib/plugin/plugin_descriptor.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import 'analytics/noop_analytics_provider.dart';
+import 'auth/anonymous_auth_plugin.dart';
+import 'auth/auth_plugin.dart';
+import 'content/content_plugin.dart';
+import 'navigation/default_navigation_plugin.dart';
+import 'navigation/navigation.dart';
+import 'network/http_network_plugin.dart';
+import 'network/network_plugin.dart';
+
+class PluginDescriptor {
```
Comment (pavanpodila) [inline packages/system/vyuh_core/lib/plugin/plugin_descriptor.dart:16]: can be final

File: packages/system/vyuh_core/lib/runtime/platform/default_platform.dart [hunk 2]
```diff
@@ -1,7 +1,39 @@
 part of '../run_app.dart';
 
+class _CacheableLookupPluginList {
```
Comment (pavanpodila) [inline packages/system/vyuh_core/lib/runtime/platform/default_platform.dart:3]: just call it `_PluginInstanceCache`

**Review by pavanpodila — CHANGES_REQUESTED** [review]

**Review by pavanpodila — APPROVED** [review]
changes done

**Comment by vishal-kr-barnwal** [comment]
Keeping it in draft, for few more chnages.

**Review by pavanpodila — APPROVED** [review]

**Comment by pavanpodila** [comment]
Fixed the developer feature and also improved the ergonomics of PluginDescriptor

**Pull request merged**